crates/services/gas_price_service/src/fuel_gas_price_updater/da_source_adapter/service.rs

```diff
@@ -0,0 +1,138 @@
+use crate::fuel_gas_price_updater::{
+    da_source_adapter::{
+        DaBlockCostsProvider,
+        POLLING_INTERVAL_MS,
+    },
+    DaBlockCosts,
+};
+use fuel_core_services::{
+    RunnableService,
+    RunnableTask,
+    ServiceRunner,
+    StateWatcher,
+};
+use std::{
+    collections::HashSet,
+    time::Duration,
+};
+use tokio::{
+    sync::mpsc::Sender,
+    time::{
+        interval,
+        Interval,
+    },
+};
+
+pub use anyhow::Result;
+
+const CACHE_COUNT: usize = 10;
+
+/// This struct houses the shared_state, polling interval
+/// and a source, which does the actual fetching of the data
+pub struct DaBlockCostsService<Source>
+where
+    Source: DaBlockCostsSource,
+{
+    block_cost_provider: DaBlockCostsProvider,
+    poll_interval: Interval,
+    source: Source,
+    sender: Sender<DaBlockCosts>,
+    cache: HashSet<DaBlockCosts>,
+}
+
+impl<Source> DaBlockCostsService<Source>
+where
+    Source: DaBlockCostsSource,
+{
+    pub fn new(source: Source, poll_interval: Option<Duration>) -> Self {
+        #[allow(clippy::arithmetic_side_effects)]
+        let (sender, receiver) = tokio::sync::mpsc::channel(
+            core::mem::size_of::<DaBlockCosts>() * CACHE_COUNT,
```

Comment:
I think the buffer is count, not memory size. Right? And maybe this `const` could be named `CHANNEL_BUFFER_SIZE`. It doesn't seem to have anything to do with the `self.cache`

Reply:
addressed in 0240ba73d